homeassistant/components/imap/coordinator.py

```diff
@@ -167,7 +167,7 @@ def _decode_payload(part: Message) -> str:
             """
             try:
                 return str(part.get_payload(decode=True).decode(self._charset))
-            except Exception:  # pylint: disable=broad-except
+            except (ValueError, TypeError):
```

Comment:
Why do we catch TypeError?

Reply:
Because the Unicode errors in the test has TypeError as a base class.

Reply:
Alternative is that we keep the general exception.

Reply:
![image](https://github.com/home-assistant/core/assets/7188918/444dd0bc-ef42-4eb4-ab98-0a36a38a3c02)

Reply:
UnicodeDecodeError is a subclass of ValueError.

Reply:
I'll have a look

Reply:
Updated to only catch `ValueError`.